### PR TITLE
Add keycloak SAML auth automation for AMG

### LINKED ISSUE #242: Unable to complete SAML auth for AMG as keycloak service is inaccessible through service Load Balancer URL
**Issue Description:**
1. The IRSA policy for AWS Load Balancer controller is missing permissions for AddTags action which is causing the NLB service association to fail.
2. The service definition for the keycloak service is using default configuration for NLB scheme leading to `internal` load balancer endpoint.
3. The keycloak version needs to be upgraded.

**Solution:**
1. Add missing permissions to AWS Load Balancer controller IRSA policy
4. Add service annotation to set the service scheme to `internet-facing`.
5. Upgrade keycloak version to latest 22.0.1

### PR DESCRIPTION
**Pull request by iamsouravin**
*Issue #, if available:* Fixes #242

*Description of changes:*
1. Add missing `AddTags` permission to AWS Load Balancer controller IRSA policy.
2. Update keycloak service annotations to create `internet-facing` NLB.
3. Upgrade keycloak version and add automation scripts for setup and cleanup.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: PetAdoptions/keycloak-setup.sh [hunk 1]
```diff
@@ -0,0 +1,305 @@
+#!/bin/bash
+
+echo "This script sets up keycloak related resources for Amazon Managed Grafana SAML authentication."
+
+export CLUSTER_NAME=PetSite
+export WORKSPACE_NAME=demo-amg
+export KEYCLOAK_NAMESPACE=keycloak
+export KEYCLOAK_REALM_AMG=amg
```
Comment (bonclay7) [inline PetAdoptions/keycloak-setup.sh:8]: We probably need to take these as inputs and fail the script if they are not provided. People can decide to name their MG workspace differently for example.

File: PetAdoptions/keycloak-cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+echo "This script cleans up keycloak related resources for Amazon Managed Grafana SAML authentication."
+
+export CLUSTER_NAME=PetSite
+export KEYCLOAK_NAMESPACE=keycloak
```
Comment (bonclay7) [inline PetAdoptions/keycloak-cleanup.sh]: Same thing here, we need to take inputs an verify them

File: PetAdoptions/keycloak-setup.sh (hunk 1 above)
Comment (bonclay7) [inline PetAdoptions/keycloak-setup.sh]: We probably need to take these as inputs and fail the script if they are not provided or use overridable default values. People can decide to name their MG workspace differently for example.

**Review by bonclay7 — CHANGES_REQUESTED** [review]
This is AMAZING! It just works with one command, thanks for doing this. I just have credentials to use and log in, perfect ❤️

We should sanitize the inputs to make it flexible and fail (or use defaults) if nothing is provided

**Comment by iamsouravin** [comment]
The scripts have been parameterized to accept the configurable fields like cluster name, workspace name, keycloak namespace, keycloak realm, etc.

**Review by bonclay7 — APPROVED** [review]
Awesome, LGTM

**Comment by bonclay7** [comment]
Closing this for duplicate

**Pull request closed without merging**